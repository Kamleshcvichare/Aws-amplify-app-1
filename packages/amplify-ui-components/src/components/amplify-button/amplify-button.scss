:host {
  --background-color: var(--amplify-amazon-orange);
  --background-color-active: var(--amplify-dark-amazon-orange);
  --background-color-disable : #ffc46d;
  --color: var(--amplify-white);
  --border-width: initial;
  --border-color: initial;
  --border-style: initial;
}

.button {
  min-width: 153px;
  display: inline-block;
  margin-bottom: 0;
  font-size: 12px;
  font-weight: 400;
  line-height: 1.42857143;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  user-select: none;
  background-image: none;
  color: var(--color);
  text-transform: uppercase;
  padding: 14px;
  letter-spacing: 1.1px;
  background-color: var(--background-color);
  border-width: var(--border-width);
  border-color: var(--border-color);
  border-style: var(--border-style);

  &:active {
    opacity: 1;
    background-color: var(--background-color-active);
  }
  &:hover {
    opacity: 0.8;
  }
  &:disabled {
    opacity: 1;
    cursor: auto;
    background-color: var(--background-color-disable);
  }
}
