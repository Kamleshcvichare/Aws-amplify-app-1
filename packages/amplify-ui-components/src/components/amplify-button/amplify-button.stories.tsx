import { storiesOf } from '@storybook/html';
import { text as textKnob } from '@storybook/addon-knobs';

storiesOf('amplify-button', module).add('with text', () => {
  const text = textKnob('Text', 'Foo');
  const type = textKnob('Type', 'button');
  return `<amplify-button type=${type}>${text}</amplify-button>`;
});
