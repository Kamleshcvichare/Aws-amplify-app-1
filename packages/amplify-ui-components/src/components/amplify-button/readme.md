# amplify-button



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default    |
| -------- | --------- | ----------- | -------- | ---------- |
| `role`   | `role`    |             | `string` | `'button'` |
| `type`   | `type`    |             | `string` | `'button'` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
