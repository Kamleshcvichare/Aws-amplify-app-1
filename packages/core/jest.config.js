module.exports = {
	...require('../../jest.config'),
	coverageThreshold: {
		global: {
			branches: 74, // todo(ashwinkumar6): temp, add tests and bump back
			functions: 72,
			lines: 87, // todo(ashwinkumar6): temp, add tests and bump back
			statements: 88,
		},
	},
};
