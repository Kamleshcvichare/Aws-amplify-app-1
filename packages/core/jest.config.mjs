import { getJestConfig } from '../../jest/getJestConfig.mjs';
import tsconfig from './tsconfig.json' assert { type: 'json' };

const jestConfig = getJestConfig(tsconfig.compilerOptions, {
	coverageThreshold: {
		global: {
			branches: 75,
			functions: 72,
			lines: 88,
			statements: 88,
		},
	},
});

export default jestConfig;
