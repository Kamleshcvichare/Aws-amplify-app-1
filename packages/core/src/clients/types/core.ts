/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
