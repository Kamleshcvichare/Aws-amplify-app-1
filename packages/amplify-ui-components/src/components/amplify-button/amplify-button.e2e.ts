import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('amplify-button', () => {
  let page: E2EPage;

  beforeEach(async () => {
    page = await newE2EPage();
  });

  it('renders', async () => {
    await page.setContent('<amplify-button></amplify-button>');
    const element = await page.find('amplify-button');
    expect(element).toHaveProperty('type');
  });

  it('fires its onButtonClicked callback upon being clicked', async () => {
    console.log("started the test");
    await page.setContent('<amplify-button></amplify-button>');

    const buttonElement = await page.find('amplify-button');
    expect(buttonElement).not.toBeNull();

    const func = jest.fn();
    await page.exposeFunction('exposedfunc', func);

    // This block here, and the .exposeFunction() above, are both necessary
    // if you ever want to pass a function into an object's props
    /*
    await page.$eval('amplify-button', (buttonElement: any) => {
      buttonElement.onButtonClicked = this.exposedfunc;
      buttonElement.type = 'submit'; // Adding a type so that the component rerenders
    });
    await page.waitForChanges();
    */

    const button = await page.find('button');
    console.log(button);
    //await button.click().catch(e => e);
    //await page.click('button');
    //await page.click('baton'); // <= fails with the error: "No node found for selector: baton" -- so page.click('button') has to be finding the button, too
    //await page.waitForChanges();
    /*
    */
   /*
    await page.focus('button');
    await page.keyboard.type('\n');
    */
   /*
    await page.$eval('button', (elem: any) => elem.click());
    */

    //expect(func).toBeCalledTimes(1);
  });
});
