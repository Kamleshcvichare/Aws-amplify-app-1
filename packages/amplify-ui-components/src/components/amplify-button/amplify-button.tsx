import { Component, Prop, h } from '@stencil/core';
import { ButtonTypes } from '../../common/types/ui-types';

@Component({
  tag: 'amplify-button',
  styleUrl: 'amplify-button.scss',
  shadow: true,
})
export class AmplifyButton {
  /** Type of the button: 'button', 'submit' or 'reset' */
  @Prop() type: ButtonTypes = 'button';
  /** (Optional) Callback called when a user clicks on the button */
  @Prop() handleButtonClick: (evt: Event) => void;
  /** (Optional) Override default styling */
  @Prop() overrideStyle: boolean = false;

  render() {
    // const emotionButtonClass = styleBranch(this.type === 'reset', button, buttonTypeReset, buttonTypeSafe);
    return (
      <button
        // class={styleNuker(this.overrideStyle, STATIC_BUTTON_CLASS_NAME, emotionButtonClass)}
        class={{
          button: true,
        }}
        type={this.type}
        onClick={this.handleButtonClick}
      >
        <slot />
      </button>
    );
  }
}
