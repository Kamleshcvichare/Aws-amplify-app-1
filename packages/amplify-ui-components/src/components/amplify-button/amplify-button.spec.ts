import { newSpecPage } from '@stencil/core/testing';
import { AmplifyButton } from './amplify-button';

describe('amplify-input', () => {
  it('renders correct HTML', async () => {
    const page = await newSpecPage({
      components: [AmplifyButton],
      html: `<amplify-button></amplify-button>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
