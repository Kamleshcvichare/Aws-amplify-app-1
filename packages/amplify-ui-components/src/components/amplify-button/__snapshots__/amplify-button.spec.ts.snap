// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-input renders with button text FOO 1`] = `
<amplify-button>
  <!---->
  <button class="amplify-ui--button css-1qhle2w" type="button">
    FOO
  </button>
</amplify-button>
`;

exports[`amplify-input renders with no button text 1`] = `
<amplify-button>
  <!---->
  <button class="amplify-ui--button css-1qhle2w" type="button"></button>
</amplify-button>
`;
